### PR TITLE
60 sofast curl interface

### PR DESCRIPTION
**Pull request by bbean23**

### CODE REVIEW

File: opencsp/app/sofast/__init__.py
Comment (braden6521) [inline opencsp/app/sofast/__init__.py:1]: ## To add:
- spatial_orientation
- DisplayShape
- DotLocations
- FacetDefinition (EnsembleDefinition if we want to support facet ensembles now)
- Surface 2D definition

File: opencsp/app/sofast/SofastServer.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        log_path_name_ext = os.path.join(log_output_dir, log_name_ext)
+        lt.logger(log_path_name_ext)
+
+    # Start the server
+    lt.warn(
+        "Warning in SofastServer: this server is unsecured. "
+        + f"It is suggested that you restrict outside access to port {port} of the host computer."
+    )
+    lt.info(f"Starting server on port {port}...")
+    server = HTTPServer(("0.0.0.0", port), SofastServer)
```
Comment (e10harvey) [inline opencsp/app/sofast/SofastServer.py:199]: Consider storing "0.0.0.0" as a local variable above near where `port` is defined.

File: opencsp/app/sofast/SofastServer.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+    # Lock in the currently allocated memory, to improve garbage collector performance
+    gc.collect()
+    gc.freeze()
+
+    # Start a new thread for the server
+    # The minimum time between server evaulation loops is determined by the GIL:
+    # https://docs.python.org/3/library/sys.html#sys.setswitchinterval
+    server_pool = ThreadPoolExecutor(max_workers=1)
+    server_pool.submit(server.serve_forever)
```
Comment (e10harvey) [inline opencsp/app/sofast/SofastServer.py:215]: What happens if this thread crashes or throws an exception? Will the program exit cleanly?

**Review by e10harvey — COMMENTED** [review]
What are the critical sections for the threads?

File: opencsp/app/sofast/SofastServer.py (hunk 1 above)
Comment (bbean23) [reply opencsp/app/sofast/SofastServer.py:199]: Yep, this is a good idea and one I will implement.

Hey, sorry about the confusion. This PR isn't ready for review yet (if I assigned you already that was a mistake). I'll let you know when it is ready.

File: opencsp/app/sofast/SofastServer.py (hunk 2 above)
Comment (bbean23) [reply opencsp/app/sofast/SofastServer.py:215]: This is a valid concern and actually why I'm doing this work. The intent is to have a separate program that restarts the entire process if the server stops responding.

**Comment by e10harvey** [comment]
Closing as won't fix.

**Pull request closed without merging**